SWLOR.Game.Server/Placeable/ScavengePoint/OnOpened.cs

```diff
@@ -106,7 +113,32 @@ public bool Run(params object[] args)
 
                     if (!string.IsNullOrWhiteSpace(spawnItem.Resref) && spawnItem.Quantity > 0)
                     {
-                        _.CreateItemOnObject(spawnItem.Resref, point.Object, spawnItem.Quantity);
+                        NWItem resource = _.CreateItemOnObject(spawnItem.Resref, point.Object, spawnItem.Quantity);
+						
+						// Add properties to the item based on Scavenging skill.  Similar logic to the resource harvester.
+						int ipBonusChance = _resource.CalculateChanceForComponentBonus(oPC, (level/10 + 1), ResourceQuality.Normal, true);
```

Comment:
It doesn't seem like there's any restriction on what items get these bonuses. That would mean it's possible for non-components to get component bonuses, I believe. They wouldn't do anything on non-components if that happens because component bonuses are separate properties from the properties which actually do the increases.

May want to do a check on the item and only apply this logic if the item spawned is a component. You should be able to pull this off with something like this:

```

var componentIP = item.ItemProperties.FirstOrDefault(x => _.GetItemPropertyType(x) == (int)CustomItemPropertyType.ComponentType);
if(componentIP != null)
    <apply these changes to the item>

```

Reply:
Applied this filtering.